public/apps/account/tenant-switch-panel.tsx

```diff
@@ -0,0 +1,251 @@
+  const [errorCallOut, setErrorCallOut] = useState<string>();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const accountInfo = (await fetchAccountInfo(props.coreStart.http)).data;
+        setTenants(accountInfo?.tenants);
+      } catch (e) {
+        // TODO: switch to better error display.
+        console.error(e);
+      }
+    };
+
+    fetchData();
+  }, [props.coreStart.http]);
+
+  // Custom tenant combo box related.
+  const [selectedCustomTenantOptions, setSelectedCustomTenantOptions] = useState<
+    EuiComboBoxOptionOption[]
+  >();
+  const onCustomTenantComboChange = (selectedOptions: EuiComboBoxOptionOption[]) => {
+    setSelectedCustomTenantOptions(selectedOptions);
+    setErrorCallOut('');
+  };
+  const customTenantOptions = _.keys(tenants).filter((tenant) => {
+    return tenant !== GLOBAL_TENANT_KEY_NAME && tenant !== props.username;
+  });
+
+  const isMultiTenancyEnabled = props.config.multitenancy.enabled;
+  const isGlobalEnabled = props.config.multitenancy.tenants.enable_global;
+  const isPrivateEnabled = props.config.multitenancy.tenants.enable_private;
+
+  const shouldDisableGlobal = !isGlobalEnabled || !tenants[GLOBAL_TENANT_KEY_NAME];
+  const getGlobalDisabledInstruction = () => {
+    if (!isGlobalEnabled) {
+      return 'Contact the administrator to enable global tenant';
+    }
+
+    if (!tenants[GLOBAL_TENANT_KEY_NAME]) {
+      return 'Contact the administrator to get access to global tenant';
+    }
+  };
+
+  // The key for private tenant is the user name.
+  const shouldDisablePrivate = !isPrivateEnabled || !tenants[props.username];
+  const getPrivateDisabledInstruction = () => {
+    if (!isPrivateEnabled) {
+      return 'Contact the administrator to enable private tenant';
+    }
+
+    if (!tenants[props.username]) {
+      return 'Contact the administrator to get access to private tenant';
+    }
+  };
+
+  // Tenant switch radios related.
+  const tenantSwitchRadios = [
+    {
+      id: GLOBAL_TENANT_RADIO_ID,
+      label: (
+        <>
+          Global
+          <EuiText size="s">The global tenant is shared between every Kibana user.</EuiText>
+          {shouldDisableGlobal && <i>{getGlobalDisabledInstruction()}</i>}
+          <EuiSpacer />
+        </>
+      ),
+      disabled: shouldDisableGlobal,
+    },
+    {
+      id: PRIVATE_TENANT_RADIO_ID,
+      label: (
+        <>
+          Private
+          <EuiText size="s">
+            The private tenant is exclusive to each user and can&apos;t be shared. You might use the
+            private tenant for exploratory work.
+          </EuiText>
+          {shouldDisablePrivate && <i>{getPrivateDisabledInstruction()}</i>}
+          <EuiSpacer />
+        </>
+      ),
+      disabled: shouldDisablePrivate,
+    },
+    {
+      id: CUSTOM_TENANT_RADIO_ID,
+      label: (
+        <>
+          Choose from custom
+          <EuiComboBox
```

Comment:
Please have a look at EuiSuperSelect which is a native single select component.

Reply:
Tried and it worked. Thanks.